tensorflow/python/ops/variable_scope.py

```diff
@@ -683,7 +683,8 @@ def _get_single_variable(self,
         init_val = initializer
         variable_dtype = None
       else:
-        if type(initializer)==type:
+        # Instantiate initializer if provided initializer is a type object.
+        if isinstance(initializer, type(init_ops.Initializer)):
           initializer=initializer(dtype)
```

Comment:
Thanks, I think it's much clearer now! Would it be possible to call with a named argument, i.e., initializer = initializer(dtype=dtype)?